pkg/component/gardener/apiserver/apiserver.go

```diff
@@ -43,6 +45,16 @@ const (
 // deleted.
 var TimeoutWaitForManagedResource = 5 * time.Minute
 
+// exposed for testing
+var (
+	// IntervalWaitServiceGardenerApiserver is the interval when waiting for
+	// gardener-apiserver service to get created by gardener-resource-manager
+	IntervalWaitServiceGardenerApiserver = 5 * time.Second
```

Comment:
There is `garden/gardener-apiserver` Service in the runtime cluster and `kube-system/gardener-apiserver` Service in the virtual cluster. I suggest to name as `IntervalWaitServiceRuntime` to properly denote for which of the Services named gardener-apiserver this const is.

```suggestion
	IntervalWaitServiceRuntime = 5 * time.Second
```